### PR TITLE
Fixes for the Qt 5.9 git branch

### PR DESCRIPTION
**Pull request by jktjkt**
Plkease check the commit changelog to evaluate their feasibility. I'm unsure about that fix which was originally done for D-Bus, but then it appears to still work on my system...

### CODE REVIEW

File: eclass/qt5-build.eclass [hunk 1]
```diff
@@ -676,9 +676,10 @@ qt5_base_configure() {
 
 	popd >/dev/null || die
 
-	if [[ ${QT5_MINOR_VERSION} -ge 8 ]]; then
+	if [[ ${QT5_MINOR_VERSION} -ge 8 && ${QT5_MINOR_VERSION} -lt 9 ]]; then
```
Comment (Pesa) [inline eclass/qt5-build.eclass]: this is always false, we don't have any 5.8 ebuilds, so you may as well delete the whole if.

Comment (jktjkt) [reply eclass/qt5-build.eclass]: Ack, done.

File: eclass/qt5-build.eclass [hunk 2]
```diff
@@ -675,12 +675,6 @@ qt5_base_configure() {
 	"${S}"/configure "${conf[@]}" || die "configure failed"
 
 	popd >/dev/null || die
-
-	if [[ ${QT5_MINOR_VERSION} -ge 8 ]]; then
```
Comment (kensington) [inline eclass/qt5-build.eclass]: Removing this block will break 5.9.1. I'm checking now to see what changed upstream in 5.9.9999

Comment (kensington) [reply eclass/qt5-build.eclass]: Live ebuilds are built out-of-source while release are not which is triggering the failure. So, we should keep this block but move it up up slightly to still operate within ``QT5_BUILD_DIR``.

File: eclass/qt5-build.eclass [hunk 3]
```diff
@@ -193,10 +193,6 @@ qt5-build_src_prepare() {
 		sed -i -e "/^QMAKE_CONF_COMPILER=/ s:=.*:=\"$(tc-getCXX)\":" \
 			configure || die "sed failed (QMAKE_CONF_COMPILER)"
 
-		# Respect toolchain and flags in config.tests
-		find config.tests/unix -name '*.test' -type f -execdir \
```
Comment (kensington) [inline eclass/qt5-build.eclass]: The upstream commit dropping the last references to -nocache is only 5.8 and later - should we make this conditional, or is it fine to nuke completely?

File: eclass/qt5-build.eclass (hunk 2 above)
Comment (Pesa) [reply eclass/qt5-build.eclass]: good catch

File: eclass/qt5-build.eclass (hunk 3 above)
Comment (Pesa) [reply eclass/qt5-build.eclass]: yep, should be made conditional.

**Comment by Pesa** [comment]
Commit message titles should start with `qt5-build.eclass:`

**Comment by kensington** [comment]
Thanks, fiddled with and picked as e44c9159b09912d55c2029d2dce478aba3ca4888..66dcf031932854324c8fbe1be883e66a66988968.

**Pull request closed without merging**